backend/src/Notifo.Domain.Integrations/Discord/DiscordIntegration.Messaging.cs

```diff
@@ -0,0 +1,101 @@
+﻿// =====================================================
+//  Notifo.io
+// ==========================================================================
+//  Copyright (c) Sebastian Stehle
+//  Author of the file: Artur Nowak
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using Discord;
+using Discord.Net;
+
+namespace Notifo.Domain.Integrations.Discord;
+public sealed partial class DiscordIntegration : IMessagingSender
+{
+    private const int Attempts = 5;
+    public const string DiscordChatId = nameof(DiscordChatId);
+
+    public void AddTargets(IDictionary<string, string> targets, UserInfo user)
+    {
+        var userId = GetUserId(user);
+
+        if (!string.IsNullOrWhiteSpace(userId))
+        {
+            targets[DiscordChatId] = userId;
+        }
+    }
+
+    public async Task<DeliveryResult> SendAsync(IntegrationContext context, MessagingMessage message,
+    CancellationToken ct)
+    {
+        if (!message.Targets.TryGetValue(DiscordChatId, out var chatId))
+        {
+            return DeliveryResult.Skipped();
+        }
+
+        return await SendMessageAsync(context, message, chatId, ct);
+    }
+
+    private async Task<DeliveryResult> SendMessageAsync(IntegrationContext context, MessagingMessage message, string chatId,
+       CancellationToken ct)
+    {
+        var botToken = BotToken.GetString(context.Properties);
+
+        for (var i = 1; i <= Attempts; i++)
+        {
+            try
+            {
+                var client = discordBotClientPool.GetDiscordClient(botToken, ct);
+
+                var user = await client.GetUserAsync(ulong.Parse(chatId));
+                if (user is null)
+                {
+                    throw new InvalidOperationException("User not found.");
+                }
+
+                EmbedBuilder builder = new EmbedBuilder();
+
+                builder.WithTitle(message.Text);
+                builder.WithDescription(message.DetailedBodyText);
+
+                if (!string.IsNullOrWhiteSpace(message.ImageSmall))
```

Comment:
What kind of image is expected here? It might make sense to provide the image in a certain format (e.g. size). The image endpoint accepts ?width and ?height query parameters.

Reply:
This image here can be of various formats and sizes - if it's needed (e.g. image is too big), Discord will do the resizing on its own.
Of course if you want, I can use the image endpoint to resize the image, but do you think it's needed?

Reply:
No, thats perfect :)